akka-actor/src/main/scala/akka/util/ByteString.scala

```diff
@@ -164,11 +183,17 @@ object ByteString {
 
   }
 
+  /** INTERNAL API: ByteString backed by exactly one array, with start / end markers */
   private[akka] object ByteString1 extends Companion {
     val empty: ByteString1 = new ByteString1(Array.empty[Byte])
-    def apply(bytes: Array[Byte]): ByteString1 = ByteString1(bytes, 0, bytes.length)
+    def fromString(s: String): ByteString1 = apply(s.getBytes)
+    def apply(bytes: Array[Byte]): ByteString1 = apply(bytes, 0, bytes.length)
     def apply(bytes: Array[Byte], startIndex: Int, length: Int): ByteString1 =
-      if (length == 0) empty else new ByteString1(bytes, startIndex, length)
+      if (length == 0) empty
+      else new ByteString1(bytes, lowerBoundToZero(startIndex), lowerBoundToZero(length))
+
+    private final def lowerBoundToZero(i: Int): Int =
+      if (i <= 0) 0 else i
```

Comment:
use `java.lang.Math.max(0, length)` since it is more likely to be intrinsified (i.e. remove lowerBoundtoZero)